irohad/ametsuchi/impl/postgres_block_storage_factory.cpp

```diff
@@ -22,15 +22,24 @@ PostgresBlockStorageFactory::PostgresBlockStorageFactory(
 std::unique_ptr<BlockStorage> PostgresBlockStorageFactory::create() {
   soci::session sql(*pool_wrapper_->connection_pool_);
   auto table = table_name_provider_();
+  if (not createTable(sql, table, log_)) {
+    return nullptr;
+  }
+  return std::make_unique<PostgresTemporaryBlockStorage>(
+      pool_wrapper_, block_factory_, std::move(table), log_);
+}
+
+bool PostgresBlockStorageFactory::createTable(soci::session &sql,
```

Comment:
why not Result<void, std::string>?

Reply:
Bump. Result would be a better solution because if will not be required to make an additional call `return expected::makeError("Unable to create " + persistent_table + " table");` in application.cpp for example.